In [ ]:
import pandas as pd
from IPython.display import display

# Import core modules directly; bypassing the CLI
from src.engine import SwingScreenerEngine
from src.models import RunOptions

In [4]:
# Define a small, focused universe for the demo
demo_symbols = ["TCS.NS", "RELIANCE.NS", "HDFCBANK.NS", "INFY.NS", "ITC.NS"]

# Instantiate the options exactly as the CLI parser
options = RunOptions(
    symbols=demo_symbols,
    cacheMode="auto",      
    quiet=True,            
    exportFile="demoOutput.xlsx"
)

In [6]:
# Initialize and run the engine
engine = SwingScreenerEngine()
results_df = engine.run(options)


displayColumns = [
    'symbol', 
    'grade', 
    'totalScore', 
    'profitabilityScore', 
    'valuationScore', 
    'peRatio', 
    'evToEbitda', 
    'netDebtToEbitda', 
    'flags'
]

# Display the top results cleanly
display(results_df[displayColumns].head(10))


🔍 Screening 5 stocks through Buffett/Lynch/Graham rules...


------------------------------------------------------------
  🟢 Strong Buy: 0  |  🟢 Buy: 0  |  🟡 Watchlist: 0
  Total screened: 5  |  Avg score: 50.5
------------------------------------------------------------

📊 Exported to demo_output.xlsx

JSON exported to demo_output_20260425_010308.json


,symbol,grade,totalScore,profitabilityScore,valuationScore,peRatio,evToEbitda,netDebtToEbitda,flags
Rank,,,,,,,,,
1,TCS.NS,🟠 Hold,58.3,21.6,5.9,17.637234,11.507,-0.488895,
2,ITC.NS,🟠 Hold,56.0,19.7,4.4,18.548586,13.331,-0.674208,
3,INFY.NS,🟡 Watchlist,54.4,19.7,5.9,15.384410,994.327,-0.590684,
4,HDFCBANK.NS,🟡 Watchlist,46.4,13.1,11.8,17.515062,NaN,NaN,
5,RELIANCE.NS,🔴 Avoid,37.6,5.6,7.4,22.222595,12.590,0.894753,


## A Note on Data Anomalies

Two rows in this output illustrate real-world data caveats rather than code bugs. HDFCBANK.NS shows NaN for EV/EBITDA and Net Debt/EBITDA because these multiples are not structurally compatible with financial institutions: banks do not operate on traditional EBITDA and their balance sheets are deposit-driven, so standard enterprise valuation ratios can break. INFY.NS shows an EV/EBITDA outlier (994.3), which is likely a yfinance data-quality issue (for example stale or inconsistent denominator inputs) rather than an economically meaningful valuation multiple. This is exactly why the screener uses a multi-factor composite framework with scoring guardrails instead of relying on any single raw metric; despite the outlier, INFY.NS still carries a reasonable valuationScore (~5.9), indicating the composite system is absorbing noisy data as intended.